### PR TITLE
nixos/uvcvideo.dynctrl: Init

### PR DESCRIPTION
**Pull request by jraygauthier**
###### Motivation for this change

Needed a way to register scientific camera's uvcvideo dynamic controls to kernel driver.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jraygauthier** [comment]
An upcoming example of dynamic controls: #41738

File: nixos/modules/hardware/video/uvcvideo/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+          Note that enabling this brings the <command>uvcdynctrl</command> tool
+          into your environement and register all dynamic controls from
+          specified <command>packages</command> to the <command>uvcvideo</command> driver.
+        '';
+      };
+
+      packages = mkOption {
+        type = types.listOf types.path;
+        default = [];
```
Comment (Mic92) [inline nixos/modules/hardware/video/uvcvideo/default.nix]: Can you provide an example value here. Is the module useful at all with an empty package list? If not, please remove `[]` as default.

Comment (jraygauthier) [reply nixos/modules/hardware/video/uvcvideo/default.nix]: @Mic92 : Good idea. Done.

Comment (jraygauthier) [reply nixos/modules/hardware/video/uvcvideo/default.nix]: Indeed not of much use without a value. Removed the default as well.

**Comment by jraygauthier** [comment]
@Mic92 : Everything fine now?

**Pull request merged**